packages/eas-cli/src/commands/update/index.ts

```diff
@@ -446,15 +448,13 @@ export default class UpdatePublish extends EasCommand {
       try {
         const platforms = platformFlag === 'all' ? defaultPublishPlatforms : [platformFlag];
         const assets = await collectAssetsAsync({ inputDir: inputDir!, platforms });
-        const { uniqueUploadedAssetCount } = await uploadAssetsAsync(
-          assets,
-          (totalAssets, missingAssets) => {
-            assetSpinner.text = `Uploading assets. Finished (${
-              totalAssets - missingAssets
-            }/${totalAssets})`;
-          }
-        );
-        uploadedAssetCount = uniqueUploadedAssetCount;
+        const uploadResults = await uploadAssetsAsync(assets, (totalAssets, missingAssets) => {
+          assetSpinner.text = `Uploading assets. Finished (${
+            totalAssets - missingAssets
+          }/${totalAssets})`;
+        });
+        uploadedAssetCount = uploadResults.uniqueUploadedAssetCount;
+        assetLimitPerUpdateGroup = uploadResults.assetLimitPerUpdateGroup;
```

Comment:
I know this is not the scope of this PR but this file needs some refactoring. The `runAsync` function is 400 LOC.
I'm mentioning this because you declared `let assetLimitPerUpdateGroup = 0;` over 100 lines above :D

Reply:
https://linear.app/expo/issue/ENG-5926/refactor-update-command